PokerGame/PokerEngine/Poker.cs

```diff
@@ -34,59 +53,42 @@ private static List<Card> CalculateHandWithBestTwoPair(List<Card> handOne, List<
             var handTwoHasTwoPair = HasTwoPair(handTwo, out handTwoRankOfFirstPair, out handTwoRankOfSeccondPair);
             var handTwoHighestPairRank = CalculateHighestPair(handTwoRankOfFirstPair, handTwoRankOfSeccondPair);
 
-            if (handOneHasTwoPair && handTwoHasTwoPair)
-            {
-                if (handOneHighestPairRank > handTwoHighestPairRank)
-                {
-                    return handOne;
-                }
-                if (handOneHighestPairRank < handTwoHighestPairRank)
-                {
-                    return handTwo;
-                }
-                return null;
-            }
-
-            if (handOneHasTwoPair)
-            {
-                return handOne;
-            }
-            if (handTwoHasTwoPair)
-            {
-                return handTwo;
-            }
-            return null;
-        }
-
-        private static Rank CalculateHighestPair(Rank pairOneRank, Rank pairTwoRank)
-        {
-            var hand = new List<Card>()
-            {
-                new Card() { Rank = pairOneRank },
-                new Card() { Rank = pairTwoRank }
-            };
-            var highestPairRank = HighestCard(hand).Rank;
-            return highestPairRank;
+            return CalculatingWinningRanking(handOne, handTwo, handOneHasTwoPair, handTwoHasTwoPair, handOneHighestPairRank, handTwoHighestPairRank);
         }
 
         private static List<Card> CalculateHandWithBestPair(List<Card> handOne, List<Card> handTwo)
         {
             Rank handOnePairRank;
-            bool handOneHasPair = HasPair(handOne, out handOnePairRank);
+            var handOneHasPair = HasPair(handOne, out handOnePairRank);
 
             Rank handTwoPairRank;
-            bool handTwoHasPair = HasPair(handTwo, out handTwoPairRank);
+            var handTwoHasPair = HasPair(handTwo, out handTwoPairRank);
 
-            if (handOneHasPair || handTwoHasPair)
+            return CalculatingWinningRanking(handOne, handTwo, handOneHasPair, handTwoHasPair, handOnePairRank, handTwoPairRank);
+        }
+        
+        private static List<Card> CalculatingWinningRanking(List<Card> handOne, List<Card> handTwo, bool handOneHasRanking, bool handTwoHasRanking, Rank handOneRank, Rank handTwoRank)
```

Comment:
Should be called HandWinsWhenEqualRankingButHigherRankOrHandHasRanking or something like that. Or, you know, refactor into 2 functions.

Reply:
i think i may have sorted this so it reads better, it will be in a later pull request.